UsageStatsLoader.inc.php

```diff
@@ -410,18 +410,31 @@ protected function getFileTypeFromFile($file) {
 	 * @param $page string
 	 * @param $op string
 	 * @param $args array
-	 * @return array (assocId, assocType)
+	 * @return array (assocId, assocType, representationId)
 	 */
 	protected function getAssoc($assocType, $contextPaths, $page, $op, $args) {
-		$assocId = $assocTypeToReturn = null;
+		$assocId = $assocTypeToReturn = $representationId = null;
 		switch ($assocType) {
 			case ASSOC_TYPE_SUBMISSION:
+				$publicationId = null;
 				if (!isset($args[0])) break;
+				// If the operation is 'view' and the arguments count > 1
+				// the arguments must be: $submissionId/version/$publicationId.
+				// Else, it is not considered hier, as submission abstract count.
+				if ($op == 'view' && count($args) > 1) {
+					if ($args[1] !== 'version') break;
+					else if (count($args) != 3) break;
+					$publicationId = (int) $args[2];
+				}
 				$submissionId = $args[0];
 				$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
-				$submission = $submissionDao->getById($submissionId);
-				if ($submission) {
-					$assocId = $submission->getId();
+				$submissionExists = $submissionDao->exists($submissionId);
+				if ($submissionExists) {
+					if ($publicationId) {
+						$publication = Services::get('publication')->get($publicationId);
+						if (!$publication) break;
```

Comment:
Maybe do the same `exists` method here unless you're using the full `$publication`. This is the really expensive part of a query because it pulls in galleys, authors, etc.